### PR TITLE
Add tests for testing num tries for executing git remote commands

### PR DESCRIPTION
**Pull request by tjlee0909**
I had to push the previous PR without tests due to it being an emergency. Following it up with this PR.

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @tjlee0909 has signed the CLA. Thanks for the pull request!

File: test/unit/project_type/test_git.py [hunk 1]
```diff
@@ -178,3 +178,15 @@ def expect_side_effect(patterns, *args, **kwargs):
...
 
         self.assertTrue(matched_prompt, "The password prompt was not matched by pexpect")
+
+    def test_execute_raises_broken_pipe_error_on_last_try(self):
+        git_executor = _GitRemoteCommandExecutor()
+        self.patch('app.project_type.git.sleep')
+        mock_manager = self.patch('app.project_type.git.Manager')
+
+        def manager_init_side_effect(*args):
+            raise BrokenPipeError('')
```
Comment (wjdhollow) [inline test/unit/project_type/test_git.py]: I feel like there is a more concise way of telling a MagicMock to raise an exception

**Comment by wjdhollow** [comment]
:+1: with small comment

File: test/unit/project_type/test_git.py (hunk 1 above)
Comment (wjdhollow) [inline test/unit/project_type/test_git.py]: Does it work if you just do a direct assignment?

mock_manager.side_effect = BrokenPipeError('')

Comment (tjlee0909) [inline test/unit/project_type/test_git.py]: No =(. It felt a little verbose to me too, and some googling had suggested to me that what you typed should have worked.

Comment (josephharrington) [inline test/unit/project_type/test_git.py]: This works in console:
```python
from unittest.mock import MagicMock
m = MagicMock()
m.test.side_effect = BrokenPipeError
m.test()
```
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/Cellar/python3/3.4.2/Frameworks/Python.framework/Versions/3.4/lib/python3.4/unittest/mock.py", line 896, in __call__
    return _mock_self._mock_call(*args, **kwargs)
  File "/usr/local/Cellar/python3/3.4.2/Frameworks/Python.framework/Versions/3.4/lib/python3.4/unittest/mock.py", line 952, in _mock_call
    raise effect
BrokenPipeError
```

We also do exception side effects in test_project_type.py and test_autoversioning.py:
```python
self.check_output_mock.side_effect = [subprocess.CalledProcessError(1, 'fake')] 
```

**Pull request merged**